### PR TITLE
BasicAuth Model

### PR DESCRIPTION
**Pull request by zagovorichev**

### CODE REVIEW

File: models/ontology/basicauth.rdf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
```
Comment (siwane) [inline models/ontology/basicauth.rdf]: Could u move this class with api stuff related (tao/models/api). What do u think?

Comment (zagovorichev) [reply models/ontology/basicauth.rdf]: it isn't just an api feature, it's something more general

File: models/classes/auth/BasicAuthType.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    /**
+     * @return array
+     * @throws \core_kernel_persistence_Exception
+     */
+    public function getCredentials()
+    {
+        $credentials = [];
+        $credentials[] = (string)$this->getInstance()->getOnePropertyValue($this->getProperty(self::PROPERTY_LOGIN));
```
Comment (siwane) [inline models/classes/auth/BasicAuthType.php]: Could u do it in one query please

File: models/classes/auth/BasicAuthType.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        // load data from instance if provided
+        $data = [
+            self::PROPERTY_LOGIN => '',
+            self::PROPERTY_PASSWORD => '',
+        ];
+
+        $instance = $this->getInstance();
+        if ($instance && $instance->exists()) {
+            $data[self::PROPERTY_LOGIN] = (string)$instance->getOnePropertyValue($this->getProperty(self::PROPERTY_LOGIN));
+            $data[self::PROPERTY_PASSWORD] = (string)$instance->getOnePropertyValue($this->getProperty(self::PROPERTY_PASSWORD));
```
Comment (siwane) [inline models/classes/auth/BasicAuthType.php]: Could u do it in one query please

File: models/ontology/basicauth.rdf (hunk 1 above)
Comment (siwane) [reply models/ontology/basicauth.rdf]: It's ok to put it under auth folder

**Comment by zagovorichev** [comment]
@siwane Update

**Comment by zagovorichev** [comment]
@siwane update

**Pull request merged**